PRESUBMIT.py

```diff
@@ -25,6 +25,15 @@ def CheckToModifyInputApi(input_api, _output_api):
     return []
 
 
+# Check Leo variables actually exist
+def CheckLeoVariables(_, output_api):
+    try:
```

Comment:
do we need to always run this check? A better approach in presubmits is to run a check only if related files were changed. You can either check for `@brave/leo` line change in `package.json` or any other change you think should require this check to be run (like `.gn/.gni` files change in `ui/...` path or something like that).

example to check for a line change in `package.json`:
https://github.com/chromium/chromium/blob/0f031973971d9e0316ee3b021e05bb97cff9017c/PRESUBMIT.py#L4616-L4639

Reply:
So ideally, we would run this check whenever someone changes a web file (.css,.less,.sass,.scss,.js,.jsx,.ts,.tsx,.lss,.svelte), as they could accidentally use a Leo variable which doesn't exist (i.e. `--leo-red-50 instead of `--leo-color-red-50`). This is particularly prevalent with font tokens, as the differences are generally pretty subtle with just the eye (we had this happen earlier).

As you say though, this is most useful when we bump Leo, so I'd be open to just running it on package bumps - whatever you think is better :smile:

Reply:
```
So ideally, we would run this check whenever someone changes a web file (.css,.less,.sass,.scss,.js,.jsx,.ts,.tsx,.lss,.svelte)
```

if you feel confident enough, you can do that :) otherwise a check of a line in `package.json` is fine to me.

it should look like this:
```
    def _web_files_filter(affected_file):
        return input_api.FilterSourceFile(
            affected_file,
            files_to_check=[r'.+\.(css|less|sass|scss|...)$'])

    if any(input_api.AffectedSourceFiles(_web_files_filter)):
      ...
```

Reply:
I'm not checking just the one line in `package.json` but I can if you think it's worth the complexity tradeoff.

Reply:
> I'm not checking just the one line in `package.json` but I can if you think it's worth the complexity tradeoff.

I think that should be enough.